src/simulator/tests/halo_simulator_test.py

```diff
@@ -89,6 +89,88 @@ def test_simulated_reach_by_spend_with_privacy(self, mock_laplace_mechanism):
         )
         self.assertTrue(reach_point.reach(1) >= 0)
 
+    def test_form_venn_diagram_regions_with_2_publishers_and_1plus_reach(self):
+        pdf1 = PublisherData([(1, 0.01), (2, 0.02), (1, 0.04), (3, 0.05)], "pdf1")
+        pdf2 = PublisherData([(2, 0.03), (4, 0.06)], "pdf2")
+        data_set = DataSet([pdf1, pdf2], "test")
+        params = SystemParameters(
+            [0.4, 0.5], LiquidLegionsParameters(), np.random.default_rng(1)
+        )
+        privacy_tracker = PrivacyTracker()
+        halo = HaloSimulator(data_set, params, privacy_tracker)
+
+        spends = [0.04, 0.04]
+        max_freq = 1
+        expected_pub_set_by_region = {1: set([0]), 3: set([0, 1])}
+        expected_regions = {1: [1], 3: [1]}
+        pub_set_by_region, regions = halo._form_venn_diagram_regions(spends, max_freq)
+        self.assertEqual(expected_pub_set_by_region, pub_set_by_region)
+        self.assertEqual(expected_regions, regions)
+
+    def test_form_venn_diagram_regions_with_2_publishers_and_2plus_reach(self):
+        pdf1 = PublisherData([(1, 0.01), (2, 0.02), (1, 0.04), (3, 0.05)], "pdf1")
+        pdf2 = PublisherData([(2, 0.03), (4, 0.06)], "pdf2")
+        data_set = DataSet([pdf1, pdf2], "test")
+        params = SystemParameters(
+            [0.4, 0.5], LiquidLegionsParameters(), np.random.default_rng(1)
+        )
+        privacy_tracker = PrivacyTracker()
+        halo = HaloSimulator(data_set, params, privacy_tracker)
+
+        spends = [0.05, 0.08]
+        max_freq = 2
+        expected_pub_set_by_region = {
+            1: set([0]),
+            2: set([1]),
+            3: set([0, 1]),
+        }
+        expected_regions = {1: [1, 1], 2: [1, 0], 3: [0, 1]}
```

Comment:
Thanks for catching this. I was stupidly mixing frequencies and k+ reaches... my intention was to test frequencies, but as mentioned in the previous [discussion](https://github.com/world-federation-of-advertisers/planning-evaluation-framework/pull/46/files#r668171952), I will modify the code so the regions will contain k+ reaches instead of frequencies and modify the unit tests accordingly.

Reply:
Done